### DISCUSSION TITLE
iminuit working with multiple cores?

### DISCUSSION THREAD

**Discussion by mbaak** [scikit-hep/iminuit#701] [Ideas]
Hello,

Is is possible to run the iminuit optimization over multiple cores?
I looked around and think not, but perhaps (hopefully) I missed something. (I recall roofit could do this, using forks.)
If not, I was wondering if there happen to be any plans to implement this in the future?

Thanks! -Max

**Comment by HDembinski** [comment]
Hi, you can use multiple cores already. Only a tiny fraction of the computing time is spend in iminuit itself, most of the computing time is spend in computing the cost function, which is under your control as the user and you can compute it in parallel. In fact, if you already use the builtin cost functions, it is sufficient to compute the pdf or cdf of your model in parallel, as shown in our tutorial on this topic:
https://iminuit.readthedocs.io/en/stable/tutorial/numba.html

In the tutorial, we make parallelized functions from scratch with numba, but you can also use the pre-made versions here (pip install numba-stats):
https://github.com/HDembinski/numba-stats
